File: app/src/main/java/com/vladan/color_interview/viewmodel/ListViewModel.java
package com.vladan.color_interview.viewmodel;

import android.content.Context;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.vladan.color_interview.Repository.ListAndPersonRepository;
import com.vladan.color_interview.api.RetrofitService;
import com.vladan.color_interview.model.ListIds;

/**
 * Created by vladan on 8/29/2020
 */
public class ListViewModel extends ViewModel {

    private final LiveData<ListIds> mListIdsLiveData;


    public ListViewModel(Context context){
        ListAndPersonRepository repository = new ListAndPersonRepository(RetrofitService.create(context));
        mListIdsLiveData = repository.getIdList();
    }

   public LiveData<ListIds> getListIdsLiveData(){
        return mListIdsLiveData;
   }

}

File: app/src/main/java/com/vladan/color_interview/viewmodel/ListViewModel.kt
package com.vladan.color_interview.viewmodel

import android.content.Context
import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.vladan.color_interview.Repository.ListAndPersonRepository.Companion.getInstance
import com.vladan.color_interview.api.RetrofitService.create
import com.vladan.color_interview.model.ListIds

/**
 * Created by vladan on 8/29/2020
 */
class ListViewModel(context: Context?) : ViewModel() {
    val listIdsLiveData: LiveData<ListIds?>

    init {
        val repository = getInstance(create(context, null), context)
        listIdsLiveData = repository!!.idList
    }
}